File: SmartValley.Domain/Entities/AllotmentEventParticipant.cs
using SmartValley.Domain.Core;

namespace SmartValley.Domain.Entities
{
    public class AllotmentEventParticipant : Entity
    {
        private AllotmentEventParticipant()
        {
        }

        public AllotmentEventParticipant(string bid, string share, long userId, bool isCollected)
        {
            Bid = bid;
            Share = share;
            UserId = userId;
            IsCollected = isCollected;
        }
        
        public string Bid { get; private set; }

        public string Share { get; private set; }

        public long UserId { get; private set; }

        public bool IsCollected { get; private set; }

        public long AllotmentEventId { get; private set; }
    }
}

File: SmartValley.FrontEnd/src/app/services/allotment-event/allotment-event-participant.ts
import BigNumber from 'bignumber.js';
import {AllotmentEventParticipantResponse} from '../../api/allotment-events/responses/allotment-event-participant-response';

export class AllotmentEventParticipant {

  public bid: BigNumber;
  public share: BigNumber;
  public userId: number;
  public isCollected: boolean;

  constructor(bid: string,
              share: string,
              userId: number,
              isCollected: boolean) {

    this.bid = new BigNumber(bid);
    this.share = new BigNumber(share);
    this.userId = userId;
    this.isCollected = isCollected;
  }

  static create(response: AllotmentEventParticipantResponse): AllotmentEventParticipant {
    return new AllotmentEventParticipant(
      response.bid,
      response.share,
      response.userId,
      response.isCollected
    );
  }
}
